lambdas/service/app.py

```diff
@@ -367,7 +367,7 @@ def link(self, *, previous: bool, **params: str) -> Optional[str]:
                     'order': self.order,
                     'size': self.size
                 }
-            return self.self_url + '?' + urllib.parse.urlencode(params)
+            return str(self.self_url) + '?' + urllib.parse.urlencode(params)
```

Comment:
Now that self_url is a furl object, consider using furl to build the url. String compositions works, but it does open up room for errors to occur.

```
furl('www.balab.com', args={'foo': ' <-space!'})
furl('www.balab.com?foo=+%3C-space%21')
```